crates/services/upgradable-executor/src/executor.rs

```diff
@@ -133,6 +134,10 @@ impl<S, R> Executor<S, R> {
         }
     }
 
+    pub fn native_executor_version(&self) -> StateTransitionBytecodeVersion {
+        self.config.native_executor_version.unwrap_or(Self::VERSION)
```

Comment:
Does something break if we update the native code but we don't increment the version?

Reply:
No in terms of execution, but we will use WASM executor instead of native which is slower=)

Reply:
I'll word my question in a different way:
As a dev working on the executor, will I ever be prompted to bump the hard-coded `Self::VERSION`?

The block could be running different versions of the native code, but the version number is the same so it doesn't use Wasm since the version number matches what is on the block. 

Or is there something preventing committing changes to the executor without bumping the version?

Reply:
https://github.com/FuelLabs/fuel-core/issues/1835